### ISSUE TITLE
JS file (vuetify) not bundled correctly, causing 404

### ISSUE THREAD

**Issue by wlcx** [parcel-bundler/parcel#787]
🐛 bug report

When trying to use parcel on a project that uses [vuetify](https://github.com/vuetifyjs/vuetify), `vuetify.js` is not bundled into the dist folder under the filename that is inserted into index.html.  

### 🎛 Configuration (.babelrc, package.json, cli command)

Create new folder and enter it
`$ mkdir reproduce-the-bug && cd $_`
Install vuetify
`$ npm install vuetify`

Create **index.html**
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title></title>
</head>
<body>
    <script src="./node_modules/vuetify/dist/vuetify.js"></script>
</body>
</html>
```

`$ parcel index.html`

### 🤔 Expected Behavior

* `vuetify.js` would be bundled into dist under a long (hashed?) filename.
* The bundled filename would be inserted into `index.html`
* The page would load

### 😯 Current Behavior

* `vuetify.js` is not renamed in the `dist` folder
```
$ ls dist
index.html  vuetify.js  vuetify.map
```
* Although, this file is different to the referenced one in `node_modules`, so some processing has happened...

* **However**, a bundled filename is inserted into `index.html`: 
```
) cat dist/index.html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title></title>
</head>
<body>
    <script src="/dist/3f0b13654a52acc5af1e96bddb0a7840.js"></script>
</body>
</html>
```

* **If the line `"main": "dist/vuetify.js"` is deleted from `node_modules/vuetify/package.json`, it works as expected** (this took some trial and error!)
### 🔦 Context
Just trying to build my stuff ;)

### 💻 Code Sample
See above!

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |1.5.1
| Node             |v8.9.4
| npm/Yarn         |5.6.0
| Operating System |MacOS 10.13.2

**Comment by DeMoorJasper** [comment]
This appears to be related to how generatebundleName works, related code: https://github.com/parcel-bundler/parcel/blob/master/src/Asset.js#L181
Not sure why the hashed name is used inside html though

**Comment by DeMoorJasper** [comment]
On further investigating i discovered that inside Asset it uses Asset.addURLDependency(...), to include it inside html, this uses 
```Javascript
this.options.parser
      .getAsset(resolved, this.package, this.options)
      .generateBundleName();
```
Which results in the hash, as this.package is the incorrect package... (the one from index.html)

But on initial gathering of packages bundler calls Resolver.resolve(...) which does call the correct package (the one from vueify)

**Comment by roblav96** [comment]
@DeMoorJasper Where would I start in fixing this problem? I've never touched parceljs before lol

**Comment by DeMoorJasper** [comment]
@roblav96 i kinda hope it got fixed with the new resolver but i’ll look into it :)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by roblav96** [comment]
@DeMoorJasper Nvm it's all good! :D This is seriously a game changer for bundling assets!!! Great work team parcel bundler =]

**Comment by devongovett** [comment]
Closing. Should be fixed by #934.
